formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -75,7 +94,39 @@ internal class FormulaManagerImpl<Input, State, Output>(
      */
     override fun evaluate(input: Input): Evaluation<Output> {
         // TODO: assert main thread.
-        val transitionID = transitionID
+
+        var result: Evaluation<Output>? = null
+        isEvaluating = true
+        while (result == null) {
+            val lastFrame = frame
+            if (lastFrame != null && executeTransitions(lastFrame.transitionID)) {
```

Comment:
We need to update the documentation for this class with the way we are now handling transitions here